src/main/java/core/basesyntax/ArrayList.java

```
@@ -1,53 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int counter;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+        counter = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        resize();
+        values[counter] = value;
+        counter++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0) {
```

if index > size we should also throw an exception